File: Data Structures & Algorithms/trapping-rain-water/submission-10.java
class Solution {
    public int trap(int[] height) {
        if (height == null || height.length < 3) return 0;
        int maxLeft=height[0],maxRight=height[height.length-1];
        int l=1,r=height.length-2,res=0;
        while(l<=r){
            if(maxLeft<=maxRight){
                maxLeft=Math.max(maxLeft,height[l]);
                res+=maxLeft-height[l];
                l++;
            }
            else {
                maxRight=Math.max(maxRight,height[r]);
                res+=maxRight-height[r];
                r--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-10.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        mp=[0]*26
        i=j=0
        res=0
        ma=0
        while j<len(s):
            mp[ord(s[j])-ord('A')]+=1
            ma=max(ma,mp[ord(s[j])-ord('A')])
            if j-i+1<=k+ma:
                res=max(res,j-i+1)
            else:
                mp[ord(s[i])-ord('A')]-=1
                i+=1
            j+=1
        return res


